confirmation/models.py

```diff
@@ -60,9 +60,6 @@ def get_object_from_key(confirmation_key: str,
         raise ConfirmationKeyException(ConfirmationKeyException.EXPIRED)
 
     obj = confirmation.content_object
-    if hasattr(obj, "status"):
```

Comment:
You dont have to remove this code block. Lets just pass an argument called `update_status` to this function with default value set to True. You can set update_status to False when its used for support.